### PR TITLE
[WIP] Fix tests for 8.2.x, use BrowserTestBase

### PR DESCRIPTION
**Pull request by joshuataylor**
WIP.

This allows tests to run for 8.2.x for me.

Using to test against travis, please don't review yet.

### CODE REVIEW

File: domain/tests/src/Functional/DomainBrowserTestBase.php [hunk 1]
```diff
@@ -161,7 +166,10 @@ protected function drupalUserIsLoggedIn($account) {
    *   The name of the domain field used to attach to the entity.
    */
   public function addDomainToEntity($entity_type, $entity_id, $id, $field = DOMAIN_ACCESS_FIELD) {
-    if ($entity = \Drupal::entityTypeManager()->getStorage($entity_type)->load($entity_id)) {
+    if ($entity = \Drupal::entityTypeManager()
```
Comment (agentrickard) [inline domain/tests/src/Functional/DomainBrowserTestBase.php:169]: Please don't make code style changes unless they are part of the d.o. standard.

**Comment by agentrickard** [comment]
I don't understand why this is required. We're running 8.x.2 tests on Travis now.

**Comment by joshuataylor** [comment]
Right now it's a mess (which is why I said no review), but I transitioned it to use BrowserTestBase, which allows tests to run via phpunit instead of via run-test.sh.

See https://www.drupal.org/node/2735005

`https://api.drupal.org/api/drupal/core%21tests%21Drupal%21Tests%21BrowserTestBase.php/class/BrowserTestBase/8.2.x`

**Comment by agentrickard** [comment]
That link was what I needed, thanks!

**Comment by zerolab** [comment]
👍 on this.

Would you be able to update .travis.yml to run the `vendor/bin/phpunit` version for 8.2.x ?

**Comment by agentrickard** [comment]
See also https://www.drupal.org/node/2783189

**Comment by agentrickard** [comment]
I've started testing locally against 8.x.2, the current test methods still work. It would be good if new tests used BrowserTestBase.

**Comment by agentrickard** [comment]
I have restarted this conversion work.

**Pull request closed without merging**